### ISSUE TITLE
Missing $ for 1000th parameter string

### ISSUE THREAD

**Issue by theobat** [nikita-volkov/hasql-dynamic-statements#2]
So if I try a query such as:
```
dynamicallyParameterizedStatement ("SELECT 1 " <> (foldMap @([]) ("," <>) $ replicate 1001 $ params (10::Int))) Decoders.noResult
```

The 1000th parameter is correctly encoded as a dynamic value, but the in-query markup $1000 misses the $ in front :
```
...,$999,1000,$1001
```
As far as I tested it only happens for the 1000th parameter...

**Comment by theobat** [comment]
I used the following versions:
- hasql == 1.4.4 
- hasql-implicits == 0.1.0.1
- hasql-dynamic-statements == 0.2.0.1

Also, I'm on GHC 8.8.3, using stackage LTS 16.2

I've looked a the prepared statement registry code, I didn't find anything, besides the error is on the markup, not the counter, and I can't find the place where this logic stands.

**Comment by nikita-volkov** [comment]
Thanks. I'm adding a regression test and looking for fix now.

**Comment by nikita-volkov** [comment]
Reproduced this in the test.

**Comment by nikita-volkov** [comment]
Found the culprit. So typical :) The use of floating point arithmetic in the "ptr" package is to blame.

**Comment by theobat** [comment]
Do you have the file/line ? I'm curious :)

**Comment by nikita-volkov** [comment]
https://github.com/nikita-volkov/ptr/blob/10febaae83a9d88296b6c0f5de89a1c51558a097/library/Ptr/Poking.hs#L121

```
> logBase 10 1000
2.9999999999999996
```
instead of the expected `3`.

**Comment by nikita-volkov** [comment]
Fixed in 0.2.0.2
